torch/utils/cpp_extension.py

```diff
@@ -2406,10 +2404,21 @@ def _get_cuda_arch_flags(cflags: Optional[list[str]] = None) -> list[str]:
     _arch_list = os.environ.get('TORCH_CUDA_ARCH_LIST', None)
 
     # If not given, determine what's best for the GPU / CUDA version that can be found
-    if not _arch_list:
-        warnings.warn(
-            "TORCH_CUDA_ARCH_LIST is not set, all archs for visible cards are included for compilation. \n"
-            "If this is not desired, please set os.environ['TORCH_CUDA_ARCH_LIST'].")
+    # Support 'native' as a special value to automatically detect only current GPU arch
```

Comment:
from my reading in the code, `arch_list=[]` exactly does `native` (the warning says: all archs for visible cards are included); so the main difference should just be that `native` makes this wish explicit and silences the warning?
```
        logger.warning(
            "TORCH_CUDA_ARCH_LIST is not set, all archs for visible cards are included for compilation. "
            "You can silence this warning by explicitly requesting TORCH_CUDA_ARCH_LIST=native"
        )
  ```

Reply:
Per your comment offline, I will make a seperate PR where the user facing API for this will look closer to cmakes